bloodhound: Handle chronyd is-active failures

stmcginnis opened the pull request:
**Issue number:**

Closes #3551 

**Description of changes:**

Bottlerocket CIS Benchmark control 2.1.1.1 verifies time sync is enabled. For our check for chronyd service status, we run `systemctl is-active chronyd` to verify its active state.

The `systemctl is-active` command doesn't just report the state string though. If the service is not active, the return code from running this command is a 3 and the command execution from rust therefore does not report success.

This changes the way the check is performed by calling `systemctl show --property ActiveState chronyd`. This command will just report the state, and its exit status will be 0 as long as the state can be retrieved. It is not affected by what that state actually is. The string check of the output is also improved to explicitly look for the full `ActiveState` output.

**Testing done:**

Built and deployed instance prefix. Ran:

```sh
systemctl stop chronyd
apiclient report cis
```

Noted the 2.1.1.1 check reported `SKIP`, indicating manual checking was required.

Built and deployed instance with this fix included. Ran:

```sh
systemctl stop chronyd
apiclient report cis
```

Verified check now reported as `FAIL`. Then ran:

```sh
systemctl start chronyd
apiclient report cis
```

Verified reported `PASS` with service running.

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

bcressey commented on sources/bloodhound/src/bin/bottlerocket-checks/checks.rs:
```diff
@@ -285,10 +285,10 @@ impl Checker for BR02010101Checker {
 
-        check_output_contains!(
+        check_output_not_contains!(
             SYSTEMCTL_CMD,
             ["is-active", "chronyd"],
-            &["active"],
+            &["inactive"],
```
This would miss the other non-active values in `ActiveState`, like "failed".

You could do the equivalent of:
```
systemctl show --property ActiveState chronyd
```

And check that the output contains `ActiveState=active`.

stmcginnis replied:
Good call!

stmcginnis committed:
bloodhound: Improve chronyd active check

The existing check for chronyd is fine if the service is running and it
properly recognizes the state. If the service is stopped or failed
though, the check directs the user to perform a manual check due to the
way the status is being checked. The "systemctl is-active" call returns
a failure exit code, causing the check to fail to determine its state.

This changes the way the check is performed by calling "systemctl show
--property ActiveState chronyd". This command will just report the
state, and its exit status will be 0 as long as the state can be
retrieved. It is not affected by what that state actually is. The string
check of the output is also improved to explicitly look for the full
ActiveState output.

Signed-off-by: Sean McGinnis <stmcg@amazon.com>
